### ISSUE TITLE
[Testbirds Feedback] GoogleMaps

### ISSUE THREAD

**Issue by Dalanie** [Integreat/app#103]
Add GoogleMaps or OSM so that we can show location-information directly in the app.

**Comment by Lemming0815** [comment]
muss es denn auf jedem Gerät gleich aussehen?
Wie wäre es mit google-maps api (easy zu integrieren) und als fallback (wenn api nicht gefunden) gibt es einen link zur Adresse (wird dann in entsprechender app geöffnen, zB OSM wenn installiert, sonst im Browser)

**Comment by dkehne** [comment]
Unsere App ist offline verfügbar - das ist quasi unser USP. Wir können nicht erwarten das Menschen mit Fuchthintergrund in den ersten 14-30 Tagen bereits einen mobilen Internet-Vertrag abgeschlossen haben. Bitte bedenken. Das geht auch erst wenn bestimmte Formulare vorliegen. Ohne Pass/Formular kein Vertrag geschweige den Prepaid. Aktuell verlinken wir in der App auf Maps.Me als Offline-Kartensystem. Vielleicht können wir dort entsprechend weiter ansetzen.

**Comment by Lemming0815** [comment]
Klar, Daniel. Ich denke den Hinweis mit externen OSM apps sollten wir auf jeden Fall beibehalten. Evtl. sogar ein popup Dialog bei erster Kartennutzung, der die Kartennutzung kurz erklärt...
Ich denke gerade an diejenigen, die die App geladen haben und schnell mal eine Wegbeschreibung zum Amt brauchen. Wenn die jetzt erst noch gezwungen sind, eine weitere app zu installieren und (zb in Berlin) 400MB Kartenmaterial zu laden bei schlechter Internetanbindung, wird dieser Teil der App unbenutzbar. Es sollte möglich sein, auch 'schnell' mal etwas nachzuschauen. Der initiale dialog sollte entsprechend darauf hinweisen, dass das nur mit Internet möglich ist.
Wenn ein user dann entscheidet, die Karte öfter und auch offline verwenden zu wollen, gibt es entsprechend die Möglichkeit über externe OSM apps zu gehen.

PS: die google maps app speichert bereits geöffnete Kartenteile im cache. Weiß nicht, ob das in der app oder in der api passiert und ob wir das übernehmen könnten.

PS2: Es gibt auch alternativen zu Maps.Me, zB OSMAnd. Die sollten wir evtl. auch vorschlagen. Beide haben VOr- und Nachteile

**Comment by Meash** [comment]
Haben wir Zahlen, wie viele Personen dieser fallback betreffen würde?
(@dkehne)
Falls das wirklich viele sind, könnten wir als fallback vlt auch einen
Screenshot der Karte einbinden (müsste dann zusätzlich vom cms zurück
kommen).

**Comment by dkehne** [comment]
Also aktuell können wir sagen - und das halte ich für politisch wichtig - dass unsere App komplett offline und ohne Datennetz benutzt werden kann. Langfristig können wir auf jeden Fall überlegen auch ne Verlinkung einzubauen...haben wir ja bei "Wichtige Links" auch schon. Wir gehen quasi davon aus, dass unsere Zielgruppe außer an öffentlichen Hotspots gar keine Internetanbindung hat. 

Zu den Zahlen: Das hängt davon ab, welche Zielgruppe wir fokussieren. Solange wir weiterhin sagen können, dass es keine eingeschränkte Funktionaliät ohne Internet gibt ist mir egal wie wir das Thema Navigation lösen. 

Osman könnten wir aufnehmen, wäre Android Only soweit ich das sehe. Klar sind das ein paar MB (Berlin z.B. 118 MB) - ich denke das sollten wir mit aufnehmen für die Evaluationen bzw. wir könnten versuchen das in den Deutschkursen herauszufinden wer was nutzt.

**Comment by Meash** [comment]
Nimmst du das mit zum evaluieren?

**Comment by Lemming0815** [comment]
Hab mir osmdroid mal angeschaut. Hat jemand eine Ahnung, wie groß das kompiliert wird?
Kann man evtl. nur Ausschnitte von OSM Kartenmaterial laden? Es muss ja nicht der ganze Stadtrand + Umland mit allen Trampelpfaden drin sein, wenn nur 5 Orte in der Innenstadt markiert wurden...
